### PR TITLE
Vert.x Core 3.x Scenario

### PR DESCRIPTION
**Pull request by BFergerson**
* Pull request NO: https://github.com/apache/incubator-skywalking/pull/2386
* Agent repository URL: https://github.com/bfergerson/incubator-skywalking
* Testcases repository URL: https://github.com/bfergerson/agent-auto-integration-testcases

@ascrutae, I'm not sure why it's giving me an issue with this one. Is it because I have multiple spans with the same operation name?

### CODE REVIEW

**Comment by wu-sheng** [comment]
> I'm not sure why it's giving me an issue with this one. Is it because I have multiple spans with the same operation name?

I think @ascrutae , you should recheck this. Vertx test case is more complex, please review the case and make sure the test framework supports it.

**Comment by wu-sheng** [comment]
@BFergerson Is this OK ready to run?

**Comment by BFergerson** [comment]
@wu-sheng, the tests won't pass but yeah it's ready to run. It's difficult to tell what is actually failing based on the logs the validator creates. It doesn't say which span it's working on or how many it has left. It is my assumption it's failing on one of the last spans and I think it might be because I have multiple spans with the same operation name.

The logs I'm getting from the validator end with:
```
[2019-03-30 11:22:19:119] [INFO] - org.skywalking.apm.test.agent.tool.validator.assertor.SegmentAssert.spansEquals(SegmentAssert.java:50) - excepted span size is equals actual span size, begin to assert span.
[2019-03-30 11:22:19:120] [INFO] - org.skywalking.apm.test.agent.tool.validator.assertor.SegmentRefAssert.findSegmentRef(SegmentRefAssert.java:36) - ref are not equal ignore this ref. 
entry service name: 
 expected: cluster-message-receiver
 actual: local-message-receiver
[2019-03-30 11:22:19:120] [ERROR] - org.skywalking.apm.test.agent.tool.Main.main(Main.java:40) - assert failed.
segment refs: 
 expected count : 1
 actual: 0
```

**Comment by BFergerson** [comment]
@wu-sheng, I'm also testing the local messages and cluster messages for the Vert.x event bus as well as the Vert.x HTTP client/server in the same scenario. Would that be an issue?

**Comment by wu-sheng** [comment]
@BFergerson I think right now, this test tool just requires the segment perfectly match. Does the async scenario make this impossible?

> segment refs: 
 expected count : 1
 actual: 0

From here, it means, there is a span expected a ref(across thread/process), but actually, it doesn't.

@ascrutae I think we need you to move in, you know this tool much better than me :)

File: vertx-core-3.x-scenario/config/expectedData.yaml [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+  - segmentId: nq 0
+    spans:
+    - operationName: "/vertx-core-3-scenario/vertx-core/core-case"
+      operationId: 0
+      parentSpanId: -1
+      spanId: 0
+      spanLayer: Http
+      tags:
+      - {key: "http.method", value: "GET"}
+      - {key: "url", value: "http://localhost:18080/vertx-core-3-scenario/vertx-core/core-case"}
```
Comment (ascrutae) [inline vertx-core-3.x-scenario/config/expectedData.yaml]:
```suggestion
      - {key: "url", value: "http://localhost:{SERVER_OUTPUT_PORT}/vertx-core-3-scenario/vertx-core/core-case"}
```

**Comment by ascrutae** [comment]
@BFergerson B The scenario built failed and here is the build failed log of scenario
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.6.0:compile (default-compile) on project vertx-core-3.x-scenario: Compilation failure
[ERROR] /home/xin/.jenkins/workspace/skywalking-agent-automated-testing/skywalking-agent-testcases/vertx-core-3.x-scenario/src/main/java/test/apache/skywalking/testcase/vertxcore/controller/VertCoreController.java:[37,49] incompatible types: io.vertx.ext.web.Router cannot be converted to io.vertx.core.Handler<io.vertx.core.http.HttpServerRequest>
```

**Comment by wu-sheng** [comment]
@BFergerson look like vertx change its API, your test case compiling fails. So I assume your pr of test case is only related to 3.6+. You may need to adjust test case name.

Also, maybe your plugin is version sensitive, need adding vitness class to identify version. If so, suggest you change plugin package name to separate versions.

**Comment by wu-sheng** [comment]
@BFergerson Look like still failure :P

**Comment by BFergerson** [comment]
@BFergerson, yeah I was hoping that would do it :P. Looks like I'll need to make some modifications to the plugin first. If I can't get more support then I'll change the naming for more version specific.

**Comment by wu-sheng** [comment]
@BFergerson Look like two cases now? event-bus and web? What should we expect?

**Comment by BFergerson** [comment]
@wu-sheng, should I not do two scenarios in one PR? I mentioned that here (https://github.com/apache/incubator-skywalking/pull/2386#issuecomment-483091447). 

This PR contains the web and eventbus scenarios. They are split up because only eventbus needs 3.2+.

I corrected the issue with the docker image versions. Everything should work now.

**Comment by wu-sheng** [comment]
@BFergerson I just want to confirm, it is OK to have two cases in one PR.

**Review by ascrutae — APPROVED** [review]

**Pull request merged**